lua/femaco/edit.lua

```diff
@@ -90,15 +90,19 @@ local get_match_at_cursor = function()
     local match_data = parse_match(match)
     local content_range = {get_match_range(match_data.content_match)}
     local ranges = {content_range}
+    local lang = match_data.lang
     if match_data.lang_match ~= nil then
       table.insert(ranges, {get_match_range(match_data.lang_match)})
+    elseif type(lang) ~= "string" then
```

Comment:
Could this maybe be moved into `parse_match` to not have to specifics of the various formats `match` can take here?

Reply:
Hope you don't mind @leisiji but I did this and merge your fix. Thanks a lot for the help :rocket:

Reply:
> Could this maybe be moved into `parse_match` to not have to deal with the specifics of the various formats `match` can take here?

I think that it might be not so easy to put that into `parse_match`, because `_lang` has been moved into injection and the two tables merge into one table. However, `parse_match` only returns one table:

```lua
-- print(vim.inspect(query.get_matches(vim.api.nvim_get_current_buf(), 'injections')))

-- return value of old style
{ {
    _lang = {
      node = <userdata 1>
    },
    injection = {
      content = {
        node = <userdata 2>
      }
    }
  } }

-- current return value, `language` table has been moved into `injection` table 
{ {
    injection = {
      content = {
        node = <userdata 1>
      },
      language = {
        node = <userdata 2>
      }
    }
  } }
```

Reply:
I might be missing something but isn't that exactly what we take into account here?
https://github.com/AckslD/nvim-FeMaco.lua/blob/9b1d7cc4a5e773b7fec6318d4981d63bcd5eaf56/lua/femaco/edit.lua#L59

Reply:
Should now be handled better in #35